java/code/src/com/redhat/rhn/frontend/xmlrpc/system/SystemHandler.java

```diff
@@ -3805,9 +3804,24 @@ public Object getCpu(User loggedInUser, Integer sid) {
     public Long[] schedulePackageInstall(User loggedInUser, List<Integer> sids,
                                          List<Integer> packageIds, Date earliestOccurrence, Boolean allowModules) {
 
-        return schedulePackagesAction(loggedInUser, sids,
-                packageIdsToMaps(loggedInUser, packageIds), earliestOccurrence,
-                ActionFactory.TYPE_PACKAGES_UPDATE, allowModules);
+        List<ErrataOverview> errataOverviews = ErrataFactory.searchByPackageIds(packageIds);
+        List<ErrataOverview> retracted = errataOverviews.stream()
+                .filter(eo -> eo.getAdvisoryStatus() == AdvisoryStatus.RETRACTED)
+                .collect(toList());
+        List<Long> longPids = packageIds.stream().map(Integer::longValue).collect(toList());
+        List<Long> retractedPids = retracted.stream()
+                .flatMap(s -> s.getPackageIds().stream())
+                .filter(longPids::contains)
+                .distinct()
+                .collect(toList());
+        if (retracted.isEmpty()) {
+            return schedulePackagesAction(loggedInUser, sids,
+                    packageIdsToMaps(loggedInUser, packageIds), earliestOccurrence,
+                    ActionFactory.TYPE_PACKAGES_UPDATE, allowModules);
+        }
```

Comment:
I think we should consider the channels assigned to the systems (for situations that a patch is retracted in one channel, but is final in the channel assigned to the system). The  `suseServerChannelsRetractedPackagesView` view provides list of retracted package ids in given server channels, which is exactly what is needed here.